core/shared/src/main/scala/org/http4s/util/SizedSeq.scala

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2013 http4s.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.http4s.util
+
+sealed trait SizedSeq[+A] {
+  def toSeq: Seq[A]
+}
+
+sealed abstract class SizedSeqBase[+A] extends SizedSeq[A] {
+  protected val seq: Seq[A]
+  override def toSeq: Seq[A] = seq
+}
+
+// format: off
+final case class SizedSeq0[+A]() extends SizedSeqBase[A] {
```

Comment:
I think this could be a case object.

Reply:
Tried to convert it to a case object, but encountered difficulty, I may tried a wrong way?
```
case object SizedSeq0 {
  def toSeq[A]: Seq[A] = Seq.empty[A]
}
```
and `EmptyCustomLabels` will look like this:
```
case object EmptyCustomLabels {
  val labels = SizedSeq0
  val values = SizedSeq0
}
```
Then [CustomMetricsOps.fromMetricsOps](https://github.com/dj707chen/http4s/blob/fd9c6ea21968b5e92442b22ac658ca0ed9fbfd5e/core/shared/src/main/scala/org/http4s/metrics/CustomMetricsOps.scala#L162) will looks like:
```
  def fromMetricsOps[F[_]](ops: MetricsOps[F]): CustomMetricsOps[F, SizedSeq0.type] = {
```
and it got compilation error:
```
Type SizedSeq0.type does not conform to upper bound SizedSeq[String] of type parameter SL
```
Root cause is that when we define `SizedSeq0` as object, it can't extend `SizedSeq` since object can't define a type parameter which `SizedSeq` requires.

So I'm leaving it as case class for now.

Reply:
Ah, I didn't think of that.  Could do something like this:

```scala
final case class SizedSeq0[+A] private () extends SizedSeq[A] {
  val toSeq: Seq[A] = Seq.empty[A]
}

final object SizedSeq0 {
  private[this] val instance = SizedSeq0[Nothing]()

  def apply[A](): SizedSeq0[A] = instance
}
```

It would be the only `SizedSeq` that can't be created with `new`, but it would prevent us from creating multiple instances when a singleton does just fine.  I'm not sure if it's worth it.

Reply:
Made the change.

Reply:
Made it also abstract to fix the following error:
```
[error] /home/runner/work/http4s/http4s/core/shared/src/main/scala/org/http4s/util/SizedSeq.scala:24:1: error: [Http4sGeneralLinters.nonValidatingCopyConstructor] Case classes with private constructors should be abstract to prevent exposing a non-validating copy constructor
[error] final case class SizedSeq0[+A] private () extends SizedSeq[A] {
[error] ^
[error] (coreJS / scalafixAll) scalafix.sbt.ScalafixFailed: LinterError
```